cpp/roaring64map.hh

```diff
@@ -794,39 +794,91 @@ public:
     }
 
     /**
-     * Compute the negation of the roaring bitmap within a specified interval.
-     * areas outside the range are passed through unchanged.
+     * Computes the negation of the roaring bitmap within the half-open interval
+     * [min, max). Areas outside the interval are unchanged.
      */
-    void flip(uint64_t range_start, uint64_t range_end) {
-        if (range_start >= range_end) {
+    void flip(uint64_t min, uint64_t max) {
+        if (min >= max) {
+            return;
+        }
+        flipClosed(min, max - 1);
+    }
+
+    /**
+     * Computes the negation of the roaring bitmap within the closed interval
+     * [min, max]. Areas outside the interval are unchanged.
+     */
+    void flipClosed(uint32_t min, uint32_t max) {
+        auto iter = roarings.begin();
+        // Since min and max are uint32_t, highbytes(min or max) == 0. The inner
+        // bitmap we are looking for, if it exists, will be at the first slot of
+        // 'roarings'.
+        if (iter == roarings.end() || iter->first != 0) {
```

Comment:
I don't think this is correct. If I have an empty `Roaring64Map` and I call `flipClosed(0, 10)`, I would expect the range from 0 to 10 to be set. Let's add a test for this case too.

Reply:
Arrrgh. Great catch!